pkg/cmd/miscellaneous.go

```diff
@@ -344,7 +346,15 @@ func isIP(word string) bool {
 }
 
 func getPublicIP() string {
-	result, err := ExecCmdReturnOutput("bash", "-c", "curl icanhazip.com")
+	ipURL, err := url.Parse("https://api.ipify.org")
+	checkError(err)
+	params := url.Values{}
+	params.Add("format", "text")
+	ipURL.RawQuery = params.Encode()
+	resp, err := http.Get(url.QueryEscape(ipURL.String()))
```

Comment:
oh I have missed this. 
instead of 
```go
resp, err := http.Get(url.QueryEscape(ipURL.String()))
```

it should be
```go
resp, err := http.Get(ipURL.String())
```
this explains the latest issue https://github.com/gardener/gardenctl/issues/273#issuecomment-681718477